module/VuFind/src/VuFind/Controller/HoldsController.php

```diff
@@ -281,7 +305,9 @@ protected function getPickupLocationsForEdit(
         int $checkLimit = 0
     ): ?array {
         $catalog = $this->getILS();
-        $holds = $catalog->getMyHolds($patron);
+        // Get holds from cache if available:
+        $holds = $this->getCachedData($this->getCacheId($patron, 'holds'))
```

Comment:
The presumption is that any edit action gets initiated from the holds list. Otherwise we wouldn't have the correct update or cancel details remembered either. And the list is refreshed after an edit or cancel, so it shouldn't be necessary to invalidate the cache manually. Unless, of course, the user is doing something weird such as using multiple browser tabs to manage stuff, but even then the edit action redirects to the list and refreshes things.

Reply:
Okay, that makes sense to me!